src/server.rs

```diff
@@ -22,26 +22,34 @@
  */
 
 use avahi;
-use client_handler::ClientHandler;
+use client_handler::{ClientHandler, Stream};
 use config::Config;
 use gps;
 use std::io;
-use std::net::{TcpListener, TcpStream};
+use std::net::{TcpListener};
+use std::os::unix::net::{UnixListener};
 use std::rc::Rc;
 use std::sync::{Arc, Mutex};
 use std::thread;
 
 pub struct Server {
     gps: Arc<Mutex<gps::GPS>>,
-    listener: TcpListener,
+    tcp_listener: TcpListener,
+    unix_listener: Option<Arc<Mutex<UnixListener>>>,
```

Comment:
hmm.. why is `Arc` and `Mutex` only needed for this one and not the tcp listener?

Reply:
The `unix_listener` gets cloned and one reference passed to a new thread in `Server::run`. It can't be moved out of Server, because then calling `Server::run` would have to consume Server.

Reply:
Ah right, we're launching a separate thread for handling the unix socket and keeping the tcp handling in the main one. We should make it consistent and handle both in separate thread.

Again, if developer time is very limited here, I understand but then I'd expect a `FIXME` here explaining the inconsistency.